src/rust/ensogl/src/system/gpu/data/texture/class.rs

```diff
@@ -45,6 +45,109 @@ impl Drop for TextureBindGuard {
 
 
 
+// ==================
+// === Parameters ===
+// ==================
+
+/// Helper struct to specify texture parameters that need to be set when binding a texture.
+///
+/// The essential parameters that need to be set are about how the texture will be samples, i.e.,
+/// how the values of the texture are interpolated at various resolutions, and how out of bounds
+/// samples are handled.
+///
+/// For more background see: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texParameter
+///
+#[derive(Copy,Clone,Debug,Default)]
+pub struct Parameters {
+    /// Specifies the setting for the texture magnification filter (`Context::TEXTURE_MIN_FILTER`).
+    pub min_filter : MinFilterValue,
+    /// Specifies the setting for the texture minification filter (`Context::TEXTURE_MAG_FILTER`).
+    pub mag_filter : MagFilterValue,
+    /// Specifies the setting for the wrapping function for texture coordinate s
+    /// (`Context::TEXTURE_WRAP_S`).
+    pub wrap_s     : WrapValue,
+    /// Specifies the setting for the wrapping function for texture coordinate t
+    /// (`Context::TEXTURE_WRAP_T`).
+    pub wrap_t     : WrapValue,
+}
+
+impl Parameters {
+    /// Sets the context parameters in the given context.
+    pub fn set_parameters(&self, context:&Context) {
+        let target = Context::TEXTURE_2D;
+        context.tex_parameteri(target,Context::TEXTURE_MIN_FILTER,self.min_filter.0 as i32);
+        context.tex_parameteri(target,Context::TEXTURE_MIN_FILTER,self.mag_filter.0 as i32);
+        context.tex_parameteri(target,Context::TEXTURE_WRAP_S,self.wrap_s.0 as i32);
+        context.tex_parameteri(target,Context::TEXTURE_WRAP_T,self.wrap_t.0 as i32);
```

Comment:
Code using `.0` is hacky as hell and hard to understand. Please name the field accordingly, like `gl_value`. Seeing `min_filter.gl_value` is much better than seeing `min_filter.0`.